toolchains/node/node_toolchain.bzl

```diff
@@ -0,0 +1,50 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module implements the node toolchain rule.
+"""
+
+NodeInfo = provider(
+    doc = "Information about how to invoke the node binary.",
```

Comment:
nit: I think its better to refer to it as the nodejs "executable" instead of "binary" in these doc comments so as not to get confused with `nodejs_binary`